### ISSUE TITLE
Istio creates an invalid inbound listener when InterceptionMode=NONE

### ISSUE THREAD

**Issue by alandiegosantos** [istio/istio#40854]
### Bug Description

When Istio is configured with **InterceptionMode** to **NONE**, a listener is created on Envoy configuration with an empty cluster. In addition, the port used by listener conflicts with the port used by the application, causing istio-proxy to reject the configuration.

## How to reproduce:
1. Apply the following manifests:
```
kubectl apply -f - << EOF
apiVersion: v1
kind: Namespace
metadata:
  name: echoserver
  labels:
    istio-injection: enabled
---
apiVersion: v1
kind: Service
metadata:
  name: echoserver
  namespace: echoserver
spec:
  selector:
    app: echoserver
  ports:
  - protocol: TCP
    port: 8080
    targetPort: 8080
    name: http
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: echoserver
  namespace: echoserver
spec:
  selector:
    matchLabels:
      app: echoserver
  replicas: 1 # tells deployment to run 2 pods matching the template
  template:
    metadata:
      labels:
        app: echoserver
      annotations:
        sidecar.istio.io/interceptionMode: NONE
    spec:
      containers:
      - name: echoserver
        image: k8s.gcr.io/echoserver:1.4
        ports:
        - containerPort: 8080
EOF
```
2. istio-proxy rejects the configuration:
```
$ kubectl logs -n echoserver $(kubectl get pods -l app=echoserver -o custom-columns=:metadata.name --no-headers -n echoserver | head -n 1) -c istio-proxy
2022-09-07T15:20:40.328609Z	info	cache	generated new workload certificate	latency=109.405105ms ttl=23h59m59.671397866s
2022-09-07T15:20:40.328747Z	info	cache	Root cert has changed, start rotating root cert
2022-09-07T15:20:40.328801Z	info	ads	XDS: Incremental Pushing:0 ConnectedEndpoints:0 Version:
2022-09-07T15:20:40.328844Z	info	cache	returned workload trust anchor from cache	ttl=23h59m59.671157905s
2022-09-07T15:20:40.336381Z	info	xdsproxy	connected to upstream XDS server: istiod.istio-system.svc:15012
2022-09-07T15:20:40.350754Z	info	ads	ADS: new connection for node:echoserver-7f696dc84c-trpng.echoserver-1
2022-09-07T15:20:40.350897Z	info	cache	returned workload trust anchor from cache	ttl=23h59m59.64911553s
2022-09-07T15:20:40.351211Z	info	ads	SDS: PUSH request for node:echoserver-7f696dc84c-trpng.echoserver resources:1 size:1.1kB resource:ROOTCA
2022-09-07T15:20:40.398267Z	info	ads	ADS: new connection for node:echoserver-7f696dc84c-trpng.echoserver-2
2022-09-07T15:20:40.398537Z	info	cache	returned workload certificate from cache	ttl=23h59m59.601470872s
2022-09-07T15:20:40.398944Z	info	ads	SDS: PUSH request for node:echoserver-7f696dc84c-trpng.echoserver resources:1 size:4.0kB resource:default
2022-09-07T15:20:40.419823Z	error	envoy config	listener '127.0.0.1_8080' failed to bind or apply socket options: cannot bind '127.0.0.1:8080': Address already in use
2022-09-07T15:20:40.424599Z	error	envoy config	listener '0.0.0.0_8080' failed to bind or apply socket options: cannot bind '0.0.0.0:8080': Address already in use
2022-09-07T15:20:40.424836Z	warning	envoy config	gRPC config for type.googleapis.com/envoy.config.listener.v3.Listener rejected: Error adding/updating listener(s) 127.0.0.1_8080: cannot bind '127.0.0.1:8080': Address already in use
0.0.0.0_8080: cannot bind '0.0.0.0:8080': Address already in use

2022-09-07T15:20:41.155903Z	warn	Envoy proxy is NOT ready: config received from XDS server, but was rejected: cds updates: 1 successful, 0 rejected; lds updates: 0 successful, 1 rejected
2022-09-07T15:20:41.235131Z	warn	Envoy proxy is NOT ready: config received from XDS server, but was rejected: cds updates: 1 successful, 0 rejected; lds updates: 0 successful, 1 rejected
2022-09-07T15:20:42.241637Z	warn	Envoy proxy is NOT ready: config received from XDS server, but was rejected: cds updates: 1 successful, 0 rejected; lds updates: 0 successful, 1 rejected
2022-09-07T15:20:43.159221Z	warn	Envoy proxy is NOT ready: config received from XDS server, but was rejected: cds updates: 1 successful, 0 rejected; lds updates: 0 successful, 1 rejected
2022-09-07T15:20:45.156165Z	warn	Envoy proxy is NOT ready: config received from XDS server, but was rejected: cds updates: 1 successful, 0 rejected; lds updates: 0 successful, 1 rejected
```

Because of the rejection of the configuration, the pod never becomes _Ready_.

### Version

```prose
$ istioctl version
client version: 1.15.0
control plane version: 1.15.0
data plane version: 1.15.0 (2 proxies)
$ kubectl version
Client Version: v1.25.0
Kustomize Version: v4.5.7
Server Version: v1.24.0
```

### Additional Information

Target cluster context: docker-desktop

Running with the following config:

istio-namespace: istio-system
full-secrets: false
timeout (mins): 30
include: {  }
exclude: { Namespaces: kube-node-lease,kube-public,kube-system,local-path-storage }
end-time: 2022-09-07 17:34:13.703263 +0200 CEST

Cluster endpoint: https://kubernetes.docker.internal:6443
CLI version:
version.BuildInfo{Version:"1.15.0", GitRevision:"e3364ab424b70ca8ee1ca76cb0b3afb73476aaac-dirty", GolangVersion:"go1.18.5", BuildStatus:"Clean", GitTag:"1.15.0"}

The following Istio control plane revisions/versions were found in the cluster:
Revision default:
&version.MeshInfo{
    {
        Component: "pilot",
        Info:      version.BuildInfo{Version:"1.15.0", GitRevision:"e3364ab424b70ca8ee1ca76cb0b3afb73476aaac", GolangVersion:"", BuildStatus:"Clean", GitTag:"1.15.0"},
    },
}

The following proxy revisions/versions were found in the cluster:
Revision default: Versions {1.15.0}

**Comment by hzxuzhonghu** [comment]
I meet more 
```
2022-09-08T06:33:38.471715Z	error	envoy config	listener '127.0.0.1_53' failed to bind or apply socket options: cannot bind '127.0.0.1:53': Permission denied
2022-09-08T06:33:38.474348Z	error	envoy config	listener '127.0.0.1_443' failed to bind or apply socket options: cannot bind '127.0.0.1:443': Permission denied
2022-09-08T06:33:38.483208Z	error	envoy config	listener '127.0.0.1_8080' failed to bind or apply socket options: cannot bind '127.0.0.1:8080': Address already in use
2022-09-08T06:33:38.494603Z	error	envoy config	listener '127.0.0.1_80' failed to bind or apply socket options: cannot bind '127.0.0.1:80': Permission denied
2022-09-08T06:33:38.515222Z	error	envoy config	listener '0.0.0.0_8080' failed to bind or apply socket options: cannot bind '0.0.0.0:8080': Address already in use
2022-09-08T06:33:38.518129Z	warning	envoy config	gRPC config for type.googleapis.com/envoy.config.listener.v3.Listener rejected: Error adding/updating listener(s) 127.0.0.1_53: cannot bind '127.0.0.1:53': Permission denied
127.0.0.1_443: cannot bind '127.0.0.1:443': Permission denied
127.0.0.1_8080: cannot bind '127.0.0.1:8080': Address already in use
127.0.0.1_80: cannot bind '127.0.0.1:80': Permission denied
0.0.0.0_8080: cannot bind '0.0.0.0:8080': Address already in use

```

**Comment by hzxuzhonghu** [comment]
I believe None mode does not work well, not sure who is using it.

**Comment by alandiegosantos** [comment]
Actually, in my company, we are designing a mesh to handle > 3M rps based on the interception mode NONE. We rely heavily on Sidecar and EnvoyFilter resource to avoid multiple listeners. And, on our setup, all HTTP traffic is handled using the http_proxy configuration.

@hzxuzhonghu  Are you aware of any issue in the InterceptionMode NONE except permissions and conflict issues?

**Comment by alandiegosantos** [comment]
We use the Sidecar object to manage the ports that are created on each pod. Right now, we are working only with the httpProxy port.

@ramaraochavali @hzxuzhonghu We are basing our implementation in the disabled InterceptionMode. So, given that, could you answer the following questions:
1. What are the plans for this mode? Is it going to be deprecated? 
2. What are the reasons behind this decision?

Thanks in advance

**Comment by hzxuzhonghu** [comment]
This could not be deprecated, some users like you will continue to use it.

**Comment by alandiegosantos** [comment]
Closed after merging of #40855
